src/pybind/mgr/prometheus/module.py

```diff
@@ -1708,6 +1724,41 @@ def get_all_daemon_health_metrics(self) -> None:
                 self.metrics[path].set(health_metric['value'], labelvalues=(
                     health_metric['type'], daemon_name,))
 
+    def get_cephadm_container_metrics(self) -> None:
+        """
+        Populate cephadm_container_cpu_percentage and cephadm_container_memory_bytes
+        from the orchestrator.
+        """
+        if not self.orch_is_available():
+            return
+
+        try:
+            daemons = raise_if_exception(self.list_daemons())
+            for daemon in daemons:
+                # Only consider daemons running in containers
+                if getattr(daemon, "container_id", None):
+                    print(daemon);
+                    cpu = getattr(daemon, "cpu_percentage", None)
+                    mem = getattr(daemon, "memory_usage", None)
+                    cpu_val = None
+                    if cpu is not None:
+                        try:
+                            cpu_val = float(str(cpu).strip().rstrip('%'))
+                        except Exception as e:
+                            self.log.error(f"Could not parse cpu_percentage '{cpu}': {e}")
+                    if cpu is not None:
+                        self.metrics['cephadm_container_cpu_percentage'].set(
+                            cpu_val,
+                            (str(daemon.daemon_type), str(daemon.daemon_id), str(daemon.daemon_name), str(daemon.hostname))
+                        )
```

Comment:
I think this has to be moved to the try/except block as you are already doing the same check there and capturing exceptions!